addon/services/adapter.js

```diff
@@ -8,7 +8,7 @@ const {
 } = Ember;
 
 function buildUrl(adapter, url) {
-  let host = getWithDefault(adapter, 'host', 'http://localhost:4200');
+  let host = getWithDefault(adapter, 'host');
```

Comment:
This doesn't need to be this complicated. `getWithDefault(adapter, 'host', '')` is all you need.

Reply:
What happens if the app is not served from the root url (`foo.com/something`)? `namespace` doesn't begin with a `/`. Wouldn't the resulting request go to `foo.com/something/namespace/request` instead of `foo.com/namespace/request`?

Reply:
This is confusing. if `host` is undefined, `namespace` is "namespace", and `url` is "request", it would go to "/namespace/request". Isn't that what you want? If it isn't, then you need more tests, because I tried my change with your test changes and everything passed.

Reply:
Sorry for the confusingness. Yes, I see how it works now. I missed the conditional inclusion of namespace. All the cases I can think of are covered. Simple version coming up.